packages/tpetra/core/src/Tpetra_MultiVectorFiller.cpp

```diff
@@ -1,5 +1,5 @@
 #include <Tpetra_MultiVectorFiller.hpp>
```

Comment:
We should actually consider the header file itself deprecated, as well as its contents.  (We could put a `#warning` in the header file, so users who include it see the warning, even if they don't use its contents.)  This means that the .cpp file would still exist, but would look like this:
```c++
#include "TpetraCore_config.h"
#ifdef TPETRA_ENABLE_DEPRECATED_CODE
#include "Tpetra_MultiVectorFiller.hpp"
// ... rest of contents of file ...
#endif 
```

Reply:
For the header:
```

#ifndef __Tpetra_MultiVectorFiller_hpp
#define __Tpetra_MultiVectorFiller_hpp

#include "Tpetra_MultiVector.hpp"
#ifdef TPETRA_ENABLE_DEPRECATED_CODE
#warning "... this header is deprecated."
...
#else
 

#error "... this header is deprecated"
#endif 
'''

Reply:
Is that what you want?

Reply:
@csiefer2 Hm, I realized that it may not be possible to turn off `#warning` warnings, or at least that this would require a different build option than the option to turn off deprecated warnings.  Perhaps we should then omit the `#warning` pragma.  Also, the first header file to include should be `"TpetraCore_config.h"`.  It is low cost to include that and nothing else, if deprecated code is off.

Reply:
So

```
#ifndef __Tpetra_MultiVectorFiller_hpp
#define __Tpetra_MultiVectorFiller_hpp

#include "Tpetra_MultiVector.hpp"
#ifdef TPETRA_ENABLE_DEPRECATED_CODE
...
#else
#error "... this header is deprecated"
#endif 
'''